File: javascript/1266.Minimum-Time-Visiting-All-Points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function (points) {
  let time = 0;

  for (let i = 0; i < points.length - 1; i++) {
    const curX = points[i][0];
    const curY = points[i][1];
    const tarX = points[i + 1][0];
    const tarY = points[i + 1][1];

    time += Math.max(Math.abs(tarX - curX), Math.abs(tarY - curY));
  }

  return time;
};

File: python/1266.Minimum-Time-Visiting-All-Points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        time = 0

        for i in range(0, len(points) - 1):
            cur_x, cur_y = points[i]
            tar_x, tar_y = points[i + 1]

            time += max(
                abs(cur_x - tar_x),
                abs(cur_y - tar_y)
            )

        return time